packages/tpetra/core/src/Tpetra_Experimental_BlockCrsMatrix_def.hpp

```diff
@@ -2929,303 +3027,323 @@ class GetLocalDiagCopy {
       // the column Map exists (is not null).
       const map_type& srcColMap = * (srcGraph.getColMap ());
 
-      // Temporary buffer for global column indices.
-      View<GO*, HES> gblColInds;
-      {
-        GO gid = 0;
-        gblColInds = PackTraits<GO, HES>::allocateArray (gid, maxRowLength, "gids");
-      }
-
       // Pack the data for each row to send, into the 'exports' buffer.
-      for (size_type i = 0; i < numExportLIDs; ++i) {
-        const LO lclRowInd = exportLIDs[i];
-        const LO* lclColIndsRaw;
-        Scalar* valsRaw;
-        LO numEntLO;
-        // It's OK to ignore the return value, since if the calling
-        // process doesn't own that local row, then the number of
-        // entries in that row on the calling process is zero.
-        (void) src->getLocalRowView (lclRowInd, lclColIndsRaw, valsRaw, numEntLO);
-        const size_t numEnt = static_cast<size_t> (numEntLO);
-        const size_t numScalarEnt = numEnt * blockSize * blockSize;
-        View<const LO*, HES, MemoryUnmanaged> lclColInds (lclColIndsRaw, numEnt);
-        const ST* valsRawST = const_cast<const ST*> (reinterpret_cast<ST*> (valsRaw));
-        View<const ST*, HES, MemoryUnmanaged> vals (valsRawST, numScalarEnt);
-
-        // NOTE (mfh 07 Feb 2015) Since we're using the host memory
-        // space here for now, this doesn't assume UVM.  That may
-        // change in the future, if we ever start packing on device.
-        const size_t numBytesPerValue = numEnt == 0 ?
-          static_cast<size_t> (0) :
-          PackTraits<ST, HES>::packValueCount (vals(0));
-
-        // Convert column indices from local to global.
-        for (size_t j = 0; j < numEnt; ++j) {
-          gblColInds(j) = srcColMap.getGlobalElement (lclColInds(j));
-        }
-
-        // Copy the row's data into the current spot in the exports array.
-        const size_t numBytes =
-          packRowForBlockCrs<ST, LO, GO, HES> (exportsK, offset, numEnt, gblColInds,
-                                               vals, numBytesPerValue, blockSize);
-        // Keep track of how many bytes we packed.
-        offset += numBytes;
-      } // for each LID (of a row) to send
-
-      if (offset != totalNumBytes) {
-        std::ostream& err = this->markLocalErrorAndGetStream ();
-        err << "packAndPreapre: At end of method, the final offset (in bytes) "
-            << offset << " does not equal the total number of bytes packed "
-            << totalNumBytes << ".  "
-            << "Please report this bug to the Tpetra developers." << endl;
-        return;
+      { 
+        typedef Kokkos::TeamPolicy<host_exec> policy_type;
+        const auto policy = 
+          policy_type(numExportLIDs, 1, 1)
+          .set_scratch_size(0, Kokkos::PerTeam(sizeof(GO)*maxRowLength));
+        Kokkos::parallel_for
+          (policy, 
+           KOKKOS_LAMBDA(const typename policy_type::member_type &member) {
```

Comment:
See above; this is a host-only loop, so we should not use the `KOKKOS_LAMBDA` macro.